### PR TITLE
DataFormats/Common : Changes needed to compile with macOS clang and libc++

### PR DESCRIPTION
**Pull request by gartung**

### CODE REVIEW

File: DataFormats/Common/interface/RangeMap.h [hunk 1]
```diff
@@ -78,6 +77,9 @@ namespace edm {
     /// to use only comparators provided with CMSSW release.
     template<typename CMP> 
     range get(ID id, CMP comparator) const {
+#ifndef _LIBCPP_VERSION
```
Comment (Dr15Jones) [inline DataFormats/Common/interface/RangeMap.h]: Can we try removing this all together to see if it was needed at all?

Comment (gartung) [inline DataFormats/Common/interface/RangeMap.h]: I think it gets used in the main cpp file.

Comment (Dr15Jones) [inline DataFormats/Common/interface/RangeMap.h]: please try rebuilding without this using.

Comment (gartung) [inline DataFormats/Common/interface/RangeMap.h]: So completely remove the using namespace statement?

Comment (gartung) [inline DataFormats/Common/interface/RangeMap.h]: OK

**Comment by gartung** [comment]
Probably needs to be merged at the same time as 
https://github.com/cms-sw/cmssw/pull/17351

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
Try replacing `__gnu_cxx::is_sorted` with `std::is_sorted`.
http://en.cppreference.com/w/cpp/algorithm/is_sorted

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

**Comment by davidlange6** [comment]
+1
(better to avoid unneeded whitespace changes in header files in the future...)

**Comment by gartung** [comment]
@davidlange6 whitespace was a remnant of a merge. Pardon my spaces...

**Pull request merged**